typings/rematch/index.d.ts

```diff
@@ -119,7 +125,14 @@ export interface Model<S = any, SS = S> {
   state: S,
   reducers?: ModelReducers<S>,
   effects?: {
-    [key: string]: (payload: any, rootState: any) => void,
+    [key: string]: (
+      this: (
+        // ExtractRematchDispatchersFromModel<this> & // TODO: Make this work, if possible
+        { [key: string]: RematchDispatcher | RematchDispatcherAsync }
```

Comment:
this.xxx access only reducers or effects too? If it's only reducers need to remove `| RemoteDispatcherAsync`

Reply:
I don't believe there is a good reason to use `this` in a reducer, only effects. Otherwise it would lead to side effects.

Reply:
No, I mean if we can only do this.reducerName or also this.otherEffectName

Reply:
Effects should be supported as well in the typings.